sample_project/Source/sample_project/WeatherQuery/WeatherQuery.cpp

```diff
@@ -0,0 +1,139 @@
+					if (feature->GetObjectField("properties")->GetStringField("COUNTRY").Contains("United States Of America"))
+					{
+						//outfield can be set in the scene on bp_feature
+						//this loop will take each outfield set in the scene and check to see if the outfield exists
+						//if it does exist, it will return the result of the outfield associated with this feature
+						//if it does not exist, it will return and error message in the scene
+						WeatherData.stationName = feature->GetObjectField("properties")->GetStringField("STATION_NAME");
+						WeatherData.country = feature->GetObjectField("properties")->GetStringField("COUNTRY");	
+						WeatherData.skyCondition = feature->GetObjectField("properties")->GetStringField("SKY_CONDTN");
+						WeatherData.tempurature = feature->GetObjectField("properties")->GetNumberField("TEMP");
+						WeatherData.weather = feature->GetObjectField("properties")->GetStringField("WEATHER");
+						
+						//this will get the geometry or coordinates of the feature
+						auto coordinates = feature->GetObjectField("geometry")->GetArrayField("coordinates");
+						//To avoid crashes, this checks to see if the type of feature is Point, if so it will get the geometry
+						//if not, it will return an error
+						//current the only type of data supported by this sample is Point Layers, but more will be added in the future.
+						WeatherData.coordinates.longitude = coordinates[0]->AsNumber();
+						WeatherData.coordinates.latitude = coordinates[1]->AsNumber();
+						//Add the data recieved into the object and load the object into an array for use later.
+						weather.Add(WeatherData);
+					}
+				}
+			}
+		}
+	}
+}
+
+//Process the request in order to get the data
+void AWeatherQuery::ProcessWebRequest()
+{
+	weather.Empty();
+	FHttpRequestRef Request = FHttpModule::Get().CreateRequest();
+	Request->OnProcessRequestComplete().BindUObject(this, &AWeatherQuery::OnResponseRecieved);
+	Request->SetURL(webLink);
+	Request->SetVerb("Get");
+	Request->ProcessRequest();
+	bDataUpdate = true;
+}
+
+void AWeatherQuery::SendCityQuery(float X, float Y)
+{
+	FString Url = "https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode";
+	UArcGISMapComponent* MapComponent = UArcGISMapComponent::GetMapComponent(this);
+	FString APIToken = MapComponent ? MapComponent->GetAPIKey() : "";
+	FString Query;
+
+	// Set up the query 
+	FHttpRequestRef Request = FHttpModule::Get().CreateRequest();
+	Request->OnProcessRequestComplete().BindUObject(this, &AWeatherQuery::ProcessCityQueryResponse);
+	FString latLong = FString::SanitizeFloat(X) + "," + FString::SanitizeFloat(Y);
+	Query = FString::Printf(TEXT("%s/?f=json&token=%s&location=%s"), *Url, *APIToken, *latLong);
+	Request->SetURL(Query);
+	Request->SetVerb("GET");
+	Request->ProcessRequest();
+}
+
+void AWeatherQuery::ProcessCityQueryResponse(FHttpRequestPtr Request, FHttpResponsePtr Response, bool bConnectedSucessfully)
+{
+	cityName = "";
```

Comment:
```suggestion
	Request->ProcessRequest();

	cityName = "";
}

void AWeatherQuery::ProcessCityQueryResponse(FHttpRequestPtr Request, FHttpResponsePtr Response, bool bConnectedSucessfully)
{
```

Moving `cityName = ""` to SendCityQuery to invalidate the old data in case the widget tries to access the name before the server has responded.